phpBB/phpbb/ucp/controller/reset_password.php

```diff
@@ -0,0 +1,442 @@
+
+namespace phpbb\ucp\controller;
+
+use phpbb\auth\auth;
+use phpbb\config\config;
+use phpbb\controller\helper;
+use phpbb\db\driver\driver_interface;
+use phpbb\event\dispatcher;
+use phpbb\exception\http_exception;
+use phpbb\language\language;
+use phpbb\log\log_interface;
+use phpbb\passwords\manager;
+use phpbb\request\request_interface;
+use phpbb\template\template;
+use phpbb\user;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+* Handling forgotten passwords via reset password functionality
+*/
+class reset_password
+{
+	/** @var config */
+	protected $config;
+
+	/** @var driver_interface */
+	protected $db;
+
+	/** @var dispatcher */
+	protected $dispatcher;
+
+	/** @var helper */
+	protected $helper;
+
+	/** @var language */
+	protected $language;
+
+	/** @var log_interface */
+	protected $log;
+
+	/** @var manager */
+	protected $passwords_manager;
+
+	/** @var request_interface */
+	protected $request;
+
+	/** @var template */
+	protected $template;
+
+	/** @var user */
+	protected $user;
+
+	/** @var array phpBB DB table names */
+	protected $tables;
+
+	/** @var string phpBB root path */
+	protected $root_path;
+
+	/** @var string PHP extension */
+	protected $php_ext;
+
+	/**
+	 * Reset password controller constructor.
+	 *
+	 * @param config $config
+	 * @param driver_interface $db
+	 * @param dispatcher $dispatcher
+	 * @param helper $helper
+	 * @param language $language
+	 * @param log_interface $log
+	 * @param manager $passwords_manager
+	 * @param request_interface $request
+	 * @param template $template
+	 * @param user $user
+	 * @param array $tables
+	 * @param $root_path
+	 * @param $php_ext
+	 */
+	public function __construct(config $config, driver_interface $db, dispatcher $dispatcher, helper $helper,
+								language $language, log_interface $log, manager $passwords_manager,
+								request_interface $request, template $template, user $user, $tables, $root_path, $php_ext)
+	{
+		$this->config = $config;
+		$this->db = $db;
+		$this->dispatcher = $dispatcher;
+		$this->helper = $helper;
+		$this->language = $language;
+		$this->log = $log;
+		$this->passwords_manager = $passwords_manager;
+		$this->request = $request;
+		$this->template = $template;
+		$this->user = $user;
+		$this->tables = $tables;
+		$this->root_path = $root_path;
+		$this->php_ext = $php_ext;
+	}
+
+	/**
+	 * Init controller
+	 */
+	protected function init_controller()
+	{
+		$this->language->add_lang('ucp');
+
+		if (!$this->config['allow_password_reset'])
+		{
+			throw new http_exception(Response::HTTP_OK, 'UCP_PASSWORD_RESET_DISABLED', [
+				'<a href="mailto:' . htmlspecialchars($this->config['board_contact']) . '">',
+				'</a>'
+			]);
+		}
+	}
+
+	/**
+	 * Remove reset token for specified user
+	 *
+	 * @param int $user_id User ID
+	 */
+	protected function remove_reset_token(int $user_id)
+	{
+		$sql_ary = [
+			'reset_token'				=> '',
+			'reset_token_expiration'	=> 0,
+		];
+
+		$sql = 'UPDATE ' . $this->tables['users'] . '
```

Comment:
It is more of a preference thing, but you only user `users` so maybe only inject that one?